### ISSUE TITLE
Getting Started with Karpenter - helm step fails

### ISSUE THREAD

**Issue by andrewleech** [aws/karpenter#3429]
Hello, I'm following through https://karpenter.sh/v0.24.0/getting-started/getting-started-with-eksctl/ and having issues with the helm step: 
```
$ eksctl create cluster -f - <<EOF
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: ${CLUSTER_NAME}
  region: ${AWS_DEFAULT_REGION}
  version: "1.24"
  tags:
    karpenter.sh/discovery: ${CLUSTER_NAME}

vpc:
  subnets:
    private:
      ap-southeast-2a: { id: subnet-4dxxxx29 }
      ap-southeast-2b: { id: subnet-57xxxx21 }
  clusterEndpoints:
    publicAccess: false
    privateAccess: true

iam:
  withOIDC: true
  serviceAccounts:
  - metadata:
      name: karpenter
      namespace: karpenter
    roleName: ${CLUSTER_NAME}-karpenter
    attachPolicyARNs:
    - arn:aws:iam::${AWS_ACCOUNT_ID}:policy/KarpenterControllerPolicy-${CLUSTER_NAME}
    roleOnly: true

iamIdentityMappings:
- arn: "arn:aws:iam::${AWS_ACCOUNT_ID}:role/KarpenterNodeRole-${CLUSTER_NAME}"
  username: system:node:{{EC2PrivateDNSName}}
  groups:
  - system:bootstrappers
  - system:nodes

fargateProfiles:
- name: karpenter
  selectors:
  - namespace: karpenter

EOF

$ echo $KARPENTER_VERSION $CLUSTER_NAME $CLUSTER_ENDPOINT $KARPENTER_IAM_ROLE_ARN
v0.24.0 gitlab-kubernetes https://EAA312B34FED5DA178305BEC21260628.gr7.ap-southeast-2.eks.amazonaws.com arn:aws:iam::465988800904:role/gitlab-kubernetes-karpenter

$ helm version
version.BuildInfo{Version:"v3.11.1", GitCommit:"293b50c65d4d56187cd4e2f390f0ada46b4c4737", GitTreeState:"clean", GoVersion:"go1.18.10"}

$ helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace karpenter --create-namespace \
>   --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
>   --set settings.aws.clusterName=${CLUSTER_NAME} \
>   --set settings.aws.clusterEndpoint=${CLUSTER_ENDPOINT} \
>   --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
>   --set settings.aws.interruptionQueueName=${CLUSTER_NAME} \
>   --wait
Error: Kubernetes cluster unreachable: Get "http://localhost:8080/version": dial tcp 127.0.0.1:8080: connect: connection refused
```

This is running on a local linux/debian machine (not in aws itself) and the earlier steps from the guide have all been completed, however I had to adjust the `eksctl` command somwhat (as shown above) to specify our private subnets and prevent public ip use.

I'm experienced with docker but this is the first time I've used Kubernetes / Helm. 
Not sure if I'm missing some presumed knowledge, or if there's an extra setup step needed to point helm to the (remote) cluster? 
I've seen references online to a `$HOME/.kube/config` which this guide doesn't mention?

Any guidance would be highly appreciated, thanks!

**Comment by andrewleech** [comment]
Update: I got past this issue by running the following just before `helm`. This also fixes `kubectl cluster-info`:
```
aws eks --region ${AWS_DEFAULT_REGION} update-kubeconfig --name ${CLUSTER_NAME}
```

As detailed on https://aws.amazon.com/premiumsupport/knowledge-center/eks-cluster-connection/

**Comment by ellistarn** [comment]
eksctl should have updated your kubeconfig for you. Any relevant logs?

**Comment by andrewleech** [comment]
Ah ok, that's good to know!

I didn't keep logs from the eksctl command but I think one step of it might have timed out, but when I saw green ticks online on all the services I thought it had worked. 
In hindsight I did have to fix the security group I specified in the config to allow port 443 before the aws command worked - this is probably the part that failed in the original command.

Sorry for the noise /seemingly incomplete bug report :-)
